### PR TITLE
Upgrade to Svelte 4

### LINKED ISSUE #2684: Upgrade to Svelte 4
https://svelte.dev/docs/v4-migration-guide

### PR DESCRIPTION
**Pull request by ericpgreen2**
Closes #2684

This PR upgrades Svelte 3 to Svelte 4. A handful of changes were required:
- I ran the Svelte 4 [migration script](https://svelte.dev/docs/v4-migration-guide) on each of our `web` workspaces
- I upgraded our `web` dependencies to versions that support Svelte 4
    - One package `@rgossiaux/svelte-headlessui` has not yet added official support for Svelte 4. Given community members have reported that [the package does work with Svelte 4 in practice](https://github.com/rgossiaux/svelte-headlessui/issues/152), I've included an [npm override](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides) in the root `package.json`.
- I fixed a few cases where Svelte 4 is more strict than Svelte 3
    - ARIA roles are required for interactive elements
    - Default slot bindings are no longer exposed to named slots (see the `floating-element` changes)
    - The `derived()` store function no longer accepts falsy values (see the `timeseries-data-store` refactor)
- I fixed failing unit tests
    - Some were failing due to our protobuf-base64 encoding/decoding functions. Fixed by migrating to `@bufbuild/protobuf`'s base64 functions.
    - Some were failing due to components' `onMount` not getting called by `vitest`. Fixed [by following this StackOverflow post](https://stackoverflow.com/questions/76577665/vitest-and-svelte-components-onmount).

### CODE REVIEW

File: web-auth/src/components/EmailPassForm.svelte [hunk 1]
```diff
@@ -68,7 +68,7 @@
 
 <div>
   {#if showForm}
-    <div class="mt-6 mb-4 flex flex-col gap-y-4" transition:slide>
+    <div class="mt-6 mb-4 flex flex-col gap-y-4" transition:slide|global>
```
Comment (AdityaHegde) [inline web-auth/src/components/EmailPassForm.svelte:71]: We have seen issues with global transitions. Has it been fixed in svelte4? Applies to all places where this change has been made.

File: web-common/src/components/chip/removable-list-chip/RemovableListChip.svelte [hunk 2]
```diff
@@ -77,7 +77,7 @@ are details left to the consumer of the component; this component should remain
         {active}
       />
     </Chip>
-    <div slot="tooltip-content" transition:fly|local={{ duration: 100, y: 4 }}>
+    <div slot="tooltip-content" transition:fly={{ duration: 100, y: 4 }}>
```
Comment (AdityaHegde) [inline web-common/src/components/chip/removable-list-chip/RemovableListChip.svelte:80]: `|local` was added to fix issues with elements lingering. Has it been fixed in svelte 4? Applies to all instances of this change.

File: web-common/src/components/chip/removable-list-chip/RemovableListChip.svelte [hunk 3]
```diff
@@ -93,6 +93,7 @@ are details left to the consumer of the component; this component should remain
   <RemovableListMenu
     {excludeStore}
     slot="floating-element"
+    let:toggleFloatingElement
```
Comment (AdityaHegde) [inline web-common/src/components/chip/removable-list-chip/RemovableListChip.svelte:96]: Why is this needed?

File: web-common/src/components/menu/wrappers/SimpleActionMenu.svelte [hunk 4]
```diff
@@ -41,6 +41,7 @@ and the menu closes.
   <slot {handleClose} toggleMenu={toggleFloatingElement} {active} />
   <Menu
     slot="floating-element"
+    let:handleClose
```
Comment (AdityaHegde) [inline web-common/src/components/menu/wrappers/SimpleActionMenu.svelte:44]: Why is this needed? Wont it mask the variable defined above on `WithTogglableFloatingElement`?

**Review by AdityaHegde — COMMENTED** [review]
Looks good mostly, just a few concerns.

File: web-auth/src/components/EmailPassForm.svelte (hunk 1 above)
Comment (ericpgreen2) [reply web-auth/src/components/EmailPassForm.svelte:71]: In Svelte 4, [transitions are local by default](https://svelte.dev/docs/v4-migration-guide#transitions-are-local-by-default). In Svelte 3, they were global by default. So this code ensures our application behaves the same as it did with Svelte 3.

All the changes to transitions were made automatically when I ran the Svelte 4 migration script: `npx svelte-migrate@latest svelte-4`.

I agree we should be wary of global transitions, but let's consider application behavior changes in a separate PR.

File: web-common/src/components/chip/removable-list-chip/RemovableListChip.svelte (hunk 2 above)
Comment (ericpgreen2) [reply web-common/src/components/chip/removable-list-chip/RemovableListChip.svelte:80]: See above – transitions are now local by default

File: web-common/src/components/chip/removable-list-chip/RemovableListChip.svelte (hunk 3 above)
Comment (ericpgreen2) [reply web-common/src/components/chip/removable-list-chip/RemovableListChip.svelte:96]: See [the migration guide's section on default slot bindings](https://svelte.dev/docs/v4-migration-guide#default-slot-bindings).

Previously, we were using _default_ slot props (like `toggleFloatingElement`) in _named_ slots (like `slot="floating-element"`). But that's no longer allowed. Now, to migrate, I've added a _named_ slot prop `toggleFloatingElement`.

**Review by AdityaHegde — APPROVED** [review]
